contrib/sddpg/core.c

```diff
@@ -190,13 +201,16 @@ static HB_ERRCODE pgsqlConnect( SQLDDCONNECTION * pConnection, PHB_ITEM pItem )
    }
    pConnection->pSDDConn = hb_xgrab( sizeof( SDDCONN ) );
    ( ( SDDCONN * ) pConnection->pSDDConn )->pConn = pConn;
+   ( ( SDDCONN * ) pConnection->pSDDConn )->existingConnection = existingConnection;
    return HB_SUCCESS;
 }
 
 
 static HB_ERRCODE pgsqlDisconnect( SQLDDCONNECTION * pConnection )
 {
-   PQfinish( ( ( SDDCONN * ) pConnection->pSDDConn )->pConn );
+   if ( ! ( ( SDDCONN * ) pConnection->pSDDConn )->existingConnection ) {
+     PQfinish( ( ( SDDCONN * ) pConnection->pSDDConn )->pConn );
+   }
    hb_xfree( pConnection->pSDDConn );
```

Comment:
This clears a point in the comments from the other PR. But formatting and indenting needs fixing here. Please use `uncrustify` with Harbour settings for these code parts.